Source/WebKit/Platform/IPC/Connection.h

```diff
@@ -650,16 +675,36 @@ Connection::AsyncReplyID Connection::sendWithAsyncReply(T&& message, C&& complet
     return { };
 }
 
+template<typename T, typename C>
+Connection::AsyncReplyID Connection::sendWithAsyncReplyOnDispatcher(T&& message, RefCountedSerialFunctionDispatcher& dispatcher, C&& completionHandler, uint64_t destinationID, OptionSet<SendOption> sendOptions)
```

Comment:
It might take some doing to support waitForAsyncReplyAndDispatchImmediately for the returned id...

Reply:
I was thinking about that, but that thing is so weird to start, I think it would be best in a follow-up change, especially considering there's no user of `sendWithAsyncReplyOnDispatcher` yet (but hopefully that will change)

Reply:
I will add an assert in `waitForAsyncReplyAndDispatchImmediately` that the AsyncReply isn't an AsyncReplyWithDispatcher